-------------------------------------------------------------------------------
Test set: RecipeControllerTest
-------------------------------------------------------------------------------
Tests run: 10, Failures: 3, Errors: 1, Skipped: 0, Time elapsed: 1.923 s <<< FAILURE! -- in RecipeControllerTest
RecipeControllerTest.testAddRecipeFailMissingID -- Time elapsed: 0.025 s <<< FAILURE!
org.mockito.exceptions.verification.TooManyActualInvocations: 

recipeView.showErrorDialog(
    "Error: Invalid ID or Name"
);
Wanted 1 time:
-> at Recipe.RecipeView.showErrorDialog(RecipeView.java:207)
But was 2 times:
-> at RecipeControllerTest.testAddRecipeFailMissingName(RecipeControllerTest.java:88)
-> at RecipeControllerTest.testAddRecipeFailMissingID(RecipeControllerTest.java:105)


	at Recipe.RecipeView.showErrorDialog(RecipeView.java:207)
	at RecipeControllerTest.testAddRecipeFailMissingID(RecipeControllerTest.java:110)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

RecipeControllerTest.testUpdateRecipeFail -- Time elapsed: 0.004 s <<< FAILURE!
org.mockito.exceptions.verification.TooManyActualInvocations: 

recipeModel.updateRecipe(
    {"category" = "Main Course", "ingredient" = "", "cuisine" = "", "instructions" = "", "cookingTime" = "", "nutrition" = "", "id" = "", "name" = ""}
);
Wanted 1 time:
-> at Recipe.RecipeModel.updateRecipe(RecipeModel.java:99)
But was 2 times:
-> at RecipeControllerTest.testFilterRecipeFail(RecipeControllerTest.java:209)
-> at RecipeControllerTest.testUpdateRecipeFail(RecipeControllerTest.java:181)


	at Recipe.RecipeModel.updateRecipe(RecipeModel.java:99)
	at RecipeControllerTest.testUpdateRecipeFail(RecipeControllerTest.java:185)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

RecipeControllerTest.testFilterRecipeSuccess -- Time elapsed: 0.002 s <<< FAILURE!
org.mockito.exceptions.verification.TooManyActualInvocations: 

recipeModel.updateRecipe(
    {"category" = "Main Course", "ingredient" = "", "cuisine" = "", "instructions" = "", "cookingTime" = "", "nutrition" = "", "id" = "", "name" = ""}
);
Wanted 1 time:
-> at Recipe.RecipeModel.updateRecipe(RecipeModel.java:99)
But was 3 times:
-> at RecipeControllerTest.testFilterRecipeFail(RecipeControllerTest.java:209)
-> at RecipeControllerTest.testUpdateRecipeFail(RecipeControllerTest.java:181)
-> at RecipeControllerTest.testFilterRecipeSuccess(RecipeControllerTest.java:195)


	at Recipe.RecipeModel.updateRecipe(RecipeModel.java:99)
	at RecipeControllerTest.testFilterRecipeSuccess(RecipeControllerTest.java:199)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

RecipeControllerTest -- Time elapsed: 1.923 s <<< ERROR!
org.jooq.exception.DataAccessException: SQL [delete from "Recipe"]; The object is already closed [90007-232]
	at org.jooq_3.19.17.H2.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3621)
	at org.jooq.impl.Tools.translate(Tools.java:3609)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:827)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:115)
	at RecipeControllerTest.cleanDatabase(RecipeControllerTest.java:249)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1117)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed [90007-232]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.message.DbException.get(DbException.java:188)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1420)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:313)
	at org.jooq.impl.ProviderEnabledConnection.prepareStatement(ProviderEnabledConnection.java:109)
	at org.jooq.impl.SettingsEnabledConnection.prepareStatement(SettingsEnabledConnection.java:91)
	at org.jooq.impl.AbstractQuery.prepare(AbstractQuery.java:431)
	at org.jooq.impl.AbstractDMLQuery.prepare0(AbstractDMLQuery.java:994)
	at org.jooq.impl.AbstractDMLQuery.prepare(AbstractDMLQuery.java:985)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:315)
	... 6 more

